invenio_communities/assets/semantic-ui/js/invenio_communities/search/components.js

```diff
@@ -0,0 +1,409 @@
+          label={label}
+          name="versions-toggle"
+          id="versions-toggle"
+          onClick={onToggleClicked}
+          checked={isChecked}
+        />
+      </Card.Content>
+    </Card>
+  );
+};
+
+export const RDMCountComponent = ({ totalResults }) => {
+  return <Label>{totalResults.toLocaleString("en-US")}</Label>;
+};
+
+export const RDMEmptyResults = (props) => {
+  const queryString = props.queryString;
+  return (
+    <>
+      <Segment placeholder textAlign="center">
+        <Header icon>
+          <Icon name="search" />
+          "No results found!"
+        </Header>
+        {queryString && (
+          <em>
+            "Current search" "{queryString}"
+          </em>
+        )}
+        <br />
+        <Button primary onClick={() => props.resetQuery()}>
+          "Clear query"
+        </Button>
+      </Segment>
+    </>
+  );
+};
+
+export const RDMErrorComponent = ({ error }) => {
+  return (
+    <Message warning>
+      <Message.Header>
+        <Icon name="warning sign" />
+        {error.response.data.message}
+      </Message.Header>
+    </Message>
+  );
+};
+
+export function SearchItemCreators({ creators }) {
+  function getIcon(creator) {
+    let ids = _get(creator, "person_or_org.identifiers", []);
+    let creatorName = _get(creator, "person_or_org.name", "No name");
+    let firstId = ids.filter((id) => ["orcid", "ror"].includes(id.scheme))[0];
+    firstId = firstId || { scheme: "" };
+    let icon = null;
+    switch (firstId.scheme) {
+      case "orcid":
+        icon = (
+          <a
+            href={"https://orcid.org/" + `${firstId.identifier}`}
+            aria-label={`${creatorName}: ${"ORCID profile"}`}
+            title={`${creatorName}: ${"ORCID profile"}`}
+          >
+            <img
+              className="inline-id-icon"
+              src="/static/images/orcid.svg"
+              alt=""
+            />
+          </a>
+        );
+        break;
+      case "ror":
+        icon = (
+          <a
+            href={"https://ror.org/" + `${firstId.identifier}`}
+            aria-label={`${creatorName}: ${"ROR profile"}`}
+            title={`${creatorName}: ${"ROR profile"}`}
+          >
+            <img
+              className="inline-id-icon"
+              src="/static/images/ror-icon.svg"
+              alt=""
+            />
+          </a>
+        );
+        break;
+      default:
+        break;
+    }
+    return icon;
+  }
+
+  function getLink(creator) {
+    let creatorName = _get(creator, "person_or_org.name", "No name");
+    let link = (
+      <a
+        href={`/search?q=metadata.creators.person_or_org.name:"${creatorName}"`}
+        title={`${creatorName}: ${"Search"}`}
+      >
+        {creatorName}
+      </a>
+    );
+    return link;
+  }
+  return creators.map((creator, index) => (
+    <span key={index}>
+      {getIcon(creator)}
+      {getLink(creator)}
+      {index < creators.length - 1 && ";"}
+    </span>
+  ));
+}
```

Comment:
Moderate: this looks like duplication of [this](https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/utils.js#L45-L110), if so I think the best would be to find a common place to put this code

Reply:
This was discussed with @ntarocco. Even do code duplication can raise some alarms, this code could eventually diverge, so, even do right now, is effectively the same code, we should keep the gate open for it being modified either in invenio-app-rdm or invenio-communities without affecting the code in the other repository.